proposals/2024-06-05_Track_Similarity_Column.md

```diff
@@ -0,0 +1,168 @@
+# Track Similarity Column
+
+<!-- markdown-toc -->
+
+* **Owners:**
+  * Daniel Fernandes
+
+* **Implementation Status:** Not implemented
+
+* **Related Issues and PRs:**
+
+  * [Issue 5655](https://github.com/mixxxdj/mixxx/issues/5655)
+  * [Issue 7810](https://github.com/mixxxdj/mixxx/issues/7810)
+  * [Issue 7599](https://github.com/mixxxdj/mixxx/issues/7599)
+  * [Issue 9896](https://github.com/mixxxdj/mixxx/issues/9896)
+  * [Issue 12753](https://github.com/mixxxdj/mixxx/issues/12753)
+
+* **Other docs or links:**
+  * [GSoC Proposal](https://docs.google.com/document/d/1QAe8mvZFKkFnG--02kuzSagzea6_ed5g3VfjuSyAY7w/edit?usp=sharing)
+
+> Proposal to add a column in the Track Library that shows how harmonically close each track is with a Target
+> Track (say, from a deck) if both tracks are played together, at the same tempo.
+
+## Why
+
+When mixing two tracks, we usually play them both at the same BPM. This means that one track may need to be tempo
+stretched, relative to the other. When Keylock is off, the tempo stretching will result in a pitch shift.
+Because of this pitch shift, we cannot solely look at the Key column to find compatible tracks.
+We need to account for the BPM as well. The similarity column will use Key and BPM information of both tracks.
+
+### Pitfalls of the current solution
+
+We have information about Key and also the Keywheel. However, if two tracks are at different BPMs, it becomes
+difficult to use this info to predict the compatibility of tracks when they're time stretched to the same tempo.
+
+## Goals
+
+Goals and use cases for the solution as proposed in [How](#how):
+
+* Make it easier to find harmonically compatible tracks to the one playing in deck
+* Make it easier to predict the compatibility of two tracks when preparing sets
+
+### Audience
+
+Users interested in creating harmonic mixes.
+
+## Non-Goals
+
+*
+
+## How
+
+### Finding the target track
+
+This is still an **Open Question**. Here's where we are at right now:
+
+If the user does not explicitly select a target track, Mixxx may automatically select a deck, based on the
+algorithm for choosing the Sync Leader.
+
+We can provide context menu options to target other decks, or tracks from the library.
```

Comment:
I think the Preview deck would actually be an elegant solution to this. Maybe something like this:

```
if track in preview deck:
    target = track in preview deck
else if sync leader present:
    target = sync leader
else:
    no target, leave similarity
```

I don't listen to the Preview deck, but I am cool with putting tracks in it. It will also let us easily see which track is being used as the target.

Reply:
Ok, nice. We can change it at any time later.